rskj-core/src/main/java/co/rsk/mine/MinerServerImpl.java

```diff
@@ -377,7 +383,15 @@ public MinerWork updateGetWork(@Nonnull final Block block, @Nonnull final boolea
     }
 
     public void setExtraData(byte[] extraData) {
-        this.extraData = extraData;
+        RLPList decodedExtraData = RLP.decodeList(this.extraData);
+        byte[] version = decodedExtraData.get(0).getRLPData();
+        byte[] identity = decodedExtraData.get(1).getRLPData();
+        this.extraData = RLP.encodeList(version, RLP.encode(identity), RLP.encodeElement(extraData));
+    }
+
+    @VisibleForTesting
```

Comment:
I would remove this method and use the `block.getExtraData()` instead

Reply:
Once we have the test with a mined block, this will be possible.